spec/models/poll/question_spec.rb

```diff
@@ -86,4 +86,203 @@
+        question_open_ended = create(:poll_question_open)
+
+        result = Poll::Question.for_physical_votes
+
+        expect(result).to match_array [question_unique, question_multiple]
+        expect(result).not_to include question_open_ended
+      end
+    end
+  end
+
+  context "open-ended results" do
+    let(:poll) { create(:poll) }
+    let!(:question_open) { create(:poll_question_open, poll: poll) }
+
+    it "includes voters who didn't answer any questions in blank answers count" do
+      create(:poll_voter, poll: poll)
+
+      expect(question_open.open_ended_blank_answers_count).to eq 1
+      expect(question_open.open_ended_valid_answers_count).to eq 0
+    end
+
+    describe "#open_ended_valid_answers_count" do
+      it "returns 0 when there are no answers" do
+        expect(question_open.open_ended_valid_answers_count).to eq 0
+      end
+
+      it "counts answers" do
+        create(:poll_answer, question: question_open, answer: "Hello")
+        create(:poll_answer, question: question_open, answer: "Bye")
+
+        expect(question_open.open_ended_valid_answers_count).to eq 2
+      end
+    end
+
+    describe "#open_ended_blank_answers_count" do
+      let!(:another_question) { create(:poll_question, poll: poll) }
+      let!(:option_yes) { create(:poll_question_option, question: another_question, title: "Yes") }
+
+      it "counts valid participants of the poll who did not answer the open-ended question" do
+        user_1 = create(:user)
+        create(:poll_answer, question: another_question, author: user_1, option: option_yes, answer: "Yes")
+        create(:poll_answer, question: another_question, option: option_yes, answer: "Yes")
+        create(:poll_answer, question: another_question, option: option_yes, answer: "Yes")
+        create(:poll_answer, question: question_open, author: user_1, answer: "Free text")
+
+        expect(question_open.open_ended_valid_answers_count).to eq 1
+        expect(question_open.open_ended_blank_answers_count).to eq 2
+      end
+
+      it "returns 0 when there are no valid participants in the poll" do
+        expect(question_open.open_ended_blank_answers_count).to eq 0
+      end
+
+      it "counts every user one time even if they answered many questions" do
+        user_1 = create(:user)
+        another_question = create(:poll_question, poll: poll)
+        option_yes = create(:poll_question_option, question: another_question, title: "Yes")
+        option_no = create(:poll_question_option, question: another_question, title: "No")
+
+        create(:poll_answer, question: another_question, author: user_1, option: option_yes, answer: "Yes")
+        create(:poll_answer, question: another_question, author: user_1, option: option_no, answer: "No")
+
+        expect(question_open.open_ended_blank_answers_count).to eq 1
+      end
+    end
+
+    describe "percentages" do
+      it "returns 0.0 when there are no total responses" do
```

Comment:
Since we've renamed the method from "responses" to "answers":

```suggestion
      it "returns 0.0 when there aren't any answers" do
```